File: src/main/java/com/example/healthylifestyleapp/DrinkSettingActivity.java
package com.example.healthylifestyleapp;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.Switch;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class DrinkSettingActivity extends AppCompatActivity implements CompoundButton.OnCheckedChangeListener, View.OnClickListener {
     Switch simpleSwitch;
ImageView imgBackSpace;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drink_setting);
        initUI();
        initListner();

    }

    private void initListner() {
        simpleSwitch.setOnCheckedChangeListener(this);
        imgBackSpace.setOnClickListener(this);

    }

    private void initUI() {
        simpleSwitch = (Switch) findViewById(R.id.simpleSwitch);
        imgBackSpace=findViewById(R.id.imgBackSpace);

    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {

                if (isChecked) {
                    Toast.makeText(getApplicationContext(), "ON", Toast.LENGTH_SHORT).show(); // display the current state for switch's

                    // The toggle is enabled
                } else {
                    // The toggle is disabled
                    Toast.makeText(getApplicationContext(), "OFF", Toast.LENGTH_SHORT).show(); // display the current state for switch's

                }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId())
        {
            case R.id.imgBackSpace:
                Intent intentBack= new Intent(this,UserProfileActivity.class);
                startActivity(intentBack);
        }

    }
}

File: app/src/main/java/com/example/healthylifestyleapp/ui/activities/DrinkSettingActivity.kt
package com.example.healthylifestyleapp.ui.activities

import android.os.Bundle
import android.view.View
import android.widget.CompoundButton
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.healthylifestyleapp.R
import kotlinx.android.synthetic.main.activity_drink_setting.*

class DrinkSettingActivity : AppCompatActivity(), CompoundButton.OnCheckedChangeListener,
    View.OnClickListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_drink_setting)
        initialize()
    }

    private fun initListner() {
        simpleSwitch.setOnCheckedChangeListener(this)

    }

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {

        if (isChecked) {
            Toast.makeText(applicationContext, "ON", Toast.LENGTH_SHORT)
                .show() // display the current state for switch's

            // The toggle is enabled
        } else {
            // The toggle is disabled
            Toast.makeText(applicationContext, "OFF", Toast.LENGTH_SHORT)
                .show() // display the current state for switch's

        }
    }

    private fun initialize() {
        setToolbar()
        initListner()
    }

    private fun setToolbar() {
        setSupportActionBar(toolbar)
        supportActionBar?.title = getString(R.string.drinks)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onClick(v: View) {
        when (v.id) {
            android.R.id.home -> {
                onBackPressed()
            }
        }

    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

}
